### ISSUE TITLE
Change how Lists and NumpyArrays are handled at the command line

### ISSUE THREAD

**Issue by JFPerkins** [AllenInstitute/argschema#67]
Currently, the way that lists and arrays are handled at the command line is to set nargs to *, allowing 1 or more arguments for the variable at the command line. If we had a schema with an item mylist which was a list of ints, we would invoke it like so at the command line:

    example_program --mylist 1 2 3 4 5 6

This handling is nice and pretty simple, and allows (requires) spaces between elements of the argument. The downside is that for any more complicated arrangement of lists or arrays (for example a list of lists or a multidimensional array), it is impossible to set them at the command line. For example, if I had a schema with a list of lists of ints, all of the following attempts at invoking result in validation errors because the elements are not lists:

    example_program --mylist 1 2 3 4 5 6
    example_program --mylist [1 2 3] [4 5 6]
    example_program --mylist [[1 2 3] [4 5 6]]
    example_program --mylist [[1,2,3],[4,5,6]]

I'd like to propose using ast.literal eval for handling lists and arrays in the argument parser. This would result in setting arrays and lists at the command line like the last of the above examples.

**Comment by JFPerkins** [comment]
Looking for feedback on this @fcollman  @nicain @NileGraddis @dyf . Not sure how many other people are actually setting arrays/lists at the command line.

**Comment by fcollman** [comment]
I'm not familiar with ast.literal eval but if there is a way to allow this feature, I can't think of a reason not to do it.  I think the only thing we discussed being actually difficult to support well was when you have a Nested( X , many=True) field... specifying that on the command line would be kind of terrible.  Although I guess [--one 1,--one 2] wouldn't be terrible syntax... i don't know how to make argparse do it. 

So in short.. sure let's do it.

**Comment by JFPerkins** [comment]
Ok, I'll start with tackling lists/arrays in the proposed manner. The tests should helpfully illustrate all the updated behavior. As for Nested(many=True), I may open a separate issue for that.

**Comment by fcollman** [comment]
closed via #75
